lambda-notes/src/components/EditNote.js

```diff
@@ -0,0 +1,77 @@
+import React, { Component } from 'react';
+import { connect } from 'react-redux';
+import { editNote } from '../actions';
+import SideBar from './SideBar';
+
+class EditNote extends Component {
+  state = {
+    note: {...this.props.note[this.props.id]},
+  };
+  
+  updateState = e => {
+    this.setState({ 
+      note: {
+        ...this.state.note,
+        [e.target.name]: e.target.value,
+      },
+    });
+  };
+
+  save = e => {
+    e.preventDefault();
+    const edited = Object.assign({}, this.state.note);
+    this.props.editNote(edited, this.state.note.id);
+    this.props.history.push('/');
+    this.setState({
+      title: '',
+      body: '',
+    });
+  }  
+
+  render() {
+    return (
+      <div className="container">
+        <SideBar />
+        <div className="formBody">
+          <div className="formHeader">Edit Note:</div>
+          <div className="formFields">
+            <form type="submit">
+              <div className="titleContainer">
+                <input
+                  type="text"
+                  className="textField"
+                  value={this.state.note.title}
+                  name="title"
+                  onChange={this.updateState}
+                />
+              </div>
+              <div className="bodyContainer">
+                <textarea
+                  type="text"
+                  className="bodyField"
+                  value={this.state.note.body}
```

Comment:
Many people forget that edit requires actual value present and not placeholder, good job on that.